feat(babel): minify typeof window (#27530)
expo/expo: docs/pages/guides/tree-shaking.mdx

- This system does not apply to server code as environment variables are not inlined in server bundles.
- Library authors should not use `EXPO_PUBLIC_` environment variables as they only run in application code for security reasons.

## Removing server code

> SDK 51 and greater

> It's common to use `typeof window === 'undefined'` to conditionally enable/disable code for server/client environments.

Reviewer: ```suggestion
It's common to use `typeof window === 'undefined'` to conditionally enable or disable code for server and client environments.
```

> `babel-preset-expo` automatically transforms `typeof window === 'undefined` to `true` when bundling for server environments, and `false` when bundling for websites. The check is left as-is when bundling for native client environments to support apps that polyfill `window`.

Reviewer: ```suggestion
`babel-preset-expo` automatically transforms `typeof window === 'undefined` to `true` when bundling for server environments and `false` when bundling for websites. The check remains unchanged when bundling for native client environments to support apps that polyfill `window`.
```

> This transform is run in both development and production, but only removes conditional requires in production.

Reviewer: ```suggestion
This transform is run in both development and production but only removes conditional requires in production.
```

No add a pause after production here.

You can configure `babel-preset-expo` to skip the transform by passing `{ preserveTypeofWindow: false }`.

<Step label="1">

```js Input
if (typeof window === 'undefined') {
  console.log('Hello on the server!');
}
```

</Step>

<Step label="2">

> The above input code snippet is transformed to the following after `babel-preset-expo` when bundling for server environments (API routes, server rendering):

Reviewer: ```suggestion
The input code from the previous step is transformed to the following code snippet after `babel-preset-expo` when bundling for server environments (API routes, server rendering):
```

The snippet above is mentioned in the previous step and we should reference that here.
